File: qc-domain/src/main/java/com/eager/questioncloud/domain/verification/repository/EmailVerificationRepository.java
package com.eager.questioncloud.domain.verification.repository;


import com.eager.questioncloud.domain.verification.model.EmailVerification;
import com.eager.questioncloud.domain.verification.vo.EmailVerificationType;

public interface EmailVerificationRepository {
    EmailVerification get(String token, EmailVerificationType emailVerificationType);

    EmailVerification getCreateUserVerification(Long userId);

    EmailVerification getForResend(String resendToken);

    EmailVerification save(EmailVerification emailVerification);
}

File: qc-core/qc-user/qc-user-core/src/main/kotlin/com/eager/questioncloud/user/repository/EmailVerificationRepository.kt
package com.eager.questioncloud.user.repository

import com.eager.questioncloud.user.domain.EmailVerification
import com.eager.questioncloud.user.enums.EmailVerificationType

interface EmailVerificationRepository {
    fun get(token: String, emailVerificationType: EmailVerificationType): EmailVerification
    
    fun getCreateUserVerification(userId: Long): EmailVerification
    
    fun getForResend(resendToken: String): EmailVerification
    
    fun save(emailVerification: EmailVerification): EmailVerification
    
    fun update(emailVerification: EmailVerification)
}
